packages/api-format/src/codec/base/Enum.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright 2017-2018 @polkadot/api-format authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { Base } from '../types';
+
+export default class CodecEnum implements Base<number> {
```

Comment:
This takes a byte and maps it to a number. We actually really should do a TS-like enum impl for these. Not sure how that looks though. Probably pass enum as type and then do some magic?